docs/transpile/defaults_and_configuration_options.mdx

```diff
@@ -0,0 +1,150 @@
+- approximation_degree (float) - Heuristic dial used for circuit approximation (1.0=no approximation, 0.0=maximal approximation). The default value is 1.0.
+- backend (Backend)
+- backend_properties (BackendProperties)
+- basis_gates (List[str])
+- callback (Callable[[BasePass, DAGCircuit, float, PropertySet, int], Any])
+- coupling_map (CouplingMap | List[List[int]])
+- dt (float)
+- hls_config (HLSConfig)
+- ignore_backend_supplied_default_methods (bool)
+- init_method (str)
+- initial_layout (Layout | Dict | List)
+- inst_map (List[InstructionScheduleMap])
+- instruction_durations (List[Tuple[str, Iterable[int], float, Iterable[float], str]] | List[Tuple[str, Iterable[int], float, Iterable[float]]] | List[Tuple[str, Iterable[int], float, str]] | List[Tuple[str, Iterable[int], float]] | InstructionDurations)
+- layout_method (str)
+- optimization_level (int)
+- optimization_method (str)
+- output_name (str | List[str])
+- routing_method (str)
+- scheduling_method (str)
+- seed_transpiler (int)
+- target (Target)
+- timing_constraints (Dict[str, int]) –
+- translation_method (str) 
+- unitary_synthesis_method (str)
+- unitary_synthesis_plugin_config (dict)
+
+## Commonly used parameters 
+ 
+### Optimization level
+
+Set the optimization level
+
+Decomposing quantum circuits into the basis gate set of the target device and the addition of swap gates needed to match hardware topology cause an increase the depth and gate count of quantum circuits. To mitigate this increased complexity, you can set the `optimization_level`.  Setting this value calls an optmization routine that optimizes circuits by combining or eliminating gates. In some cases these methods are so effective the output circuits have lower depth than the inputs. In other cases, not much can be done, and the computation may be difficult to perform on noisy devices. 
+
+Different gate optimizations are turned on with different `optimization_level` values. In general, the higher the number, the more optimized it is, at the expense of longer processing time. Choosing the best optimization level might take trial and error, as it depends heavily on the circuit being transpiled and the backend being targeted. By default, `optimization_level=1` is used.
+
+Levels: 
+
+
+The `optimization_level` setting specifies how much optimization to perform on the circuits. Higher levels generate more optimized circuits, at the expense of longer compile times.
+
+- `optimization_level=0`: Trivial optimization, which maps the circuit to the backend with no explicit optimization (except whatever optimizations the mapper does).
+- `optimization_level=1`: Light optimization, which maps the circuit and does simple optimizations by collapsing adjacent gates. This is the default value.
+- `optimization_level=2`: Medium optimization, including a noise-adaptive layout and a gate-cancellation procedure based on gate commutation relationships.
```

Comment:
All optimization levels >= 1 are noise aware now, they actual all use the same basic layout and routing algorithms, we just use increasing the heuristic effort spent trying to optimize. There is an a doc PR on qiskit that was looking at adding a how to guide for optimization levels where we had a very similar discussion about these descriptions: https://github.com/Qiskit/qiskit/pull/10303

Reply:
So should I even mention the different processes that take place, or just describe that for each level (1-3),  the same heuristic layout and swap insertion technique is used, but with an increased number of iterations for each level?  And if that's the case, it seems silly to split them out individually in a list or table, unless it's 0 vs 1-3.

Reply:
Level 0 is using different techniques, but level 1-3 are doing exactly the same thing (with the exception of only level 1 trying trivial first before running vf2). I guess it depends on how we want to structure the documentation. For me personally the table of what exactly is run in each optimization level probably isn't relevant for this guide (I also generally worry about keeping it up to date every release) just the general trend of increasing optimization level value trades off compilation time with a potentially more optimized output circuit.

Reply:
OK, I'm going to merge those descriptions.